packages/data-provider/src/utils/auditUtils.ts

```diff
@@ -0,0 +1,51 @@
+import * as _ from 'lodash-es';
+
+import { AUDIT_ACTION, AuditAction, AuditDataRepository, AuditDataResponse } from './types.js';
+
+/**
+ * Returns `true` if input value matches with a valid Audit Event type.
+ * Otherwise returns `false`
+ * @param {unknown} value
+ * @returns {boolean}
+ */
+export const isAuditEventValid = (value: unknown): boolean => {
+	return (
+		typeof value === 'string' &&
+		[AUDIT_ACTION.DELETE.toString(), AUDIT_ACTION.UPDATE.toString()].includes(value.toUpperCase())
+	);
+};
+
+/**
+ * Convert a value string into it's Audit event type if it matches.
+ * Otherwise it returns `undefined`
+ * @param {string} value
+ * @returns {AuditAction | undefined}
+ */
+export const convertToAuditEvent = (value: string): AuditAction | undefined => {
+	if (isAuditEventValid(value)) {
+		if (value.toUpperCase() === AUDIT_ACTION.UPDATE.toString()) return AUDIT_ACTION.UPDATE;
+		else if (value.toUpperCase() === AUDIT_ACTION.DELETE.toString()) return AUDIT_ACTION.DELETE;
```

Comment:
No need for these toString() calls. AUDIT_ACTION.DELETE is already a string.

Reply:
Updated this function to check using Zod instead